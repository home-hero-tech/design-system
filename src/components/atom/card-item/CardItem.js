import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import classNames from 'classnames';

import css from './CardItem.module.scss';

import EmptyCardItem from '../empty-card-item/EmptyCardItem';

const CardItem = ({ format, label, leftIcon, value, small, ...otherProps }) => {
  const valueClasses = classNames({
    [css['c-card-item__value']]: true,
    [css['c-card-item__value--small']]: !!small,
    [css.truncate]: true
  });

  const labelClasses = classNames({
    [css['c-card-item__label']]: true,
    [css['c-card-item__label--small']]: !!small,
    [css.truncate]: true
  });

  const renderContent = (_label, _value, _leftIcon) => (
    <>
      {_leftIcon || null}
      <div className={css['c-card-item__content']}>
        {_label ? <span className={labelClasses}>{_label}</span> : null}
        {_value ? (
          <span className={valueClasses}>
            {_value instanceof Date ? moment(_value).format(format) : _value}
          </span>
        ) : null}
      </div>
    </>
  );

  return (
    <EmptyCardItem small={small} {...otherProps}>
      {renderContent(label, value, leftIcon)}
    </EmptyCardItem>
  );
};

CardItem.propTypes = {
  ellipsis: PropTypes.bool,
  format: PropTypes.string,
  gray: PropTypes.bool,
  label: PropTypes.string,
  lbPrimary: PropTypes.bool,
  lbWhite: PropTypes.bool,
  lbGray: PropTypes.bool,
  leftIcon: PropTypes.instanceOf(Object),
  primary: PropTypes.bool,
  white: PropTypes.bool,
  small: PropTypes.bool,
  svgPrimary: PropTypes.bool,
  svgWhite: PropTypes.bool,
  valPrimary: PropTypes.bool,
  valWhite: PropTypes.bool,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date),
    PropTypes.number
  ]).isRequired
};

CardItem.defaultProps = {
  ellipsis: false,
  format: 'DD MMM YYYY [Ã s] HH:mm',
  gray: false,
  label: null,
  lbPrimary: false,
  lbWhite: false,
  lbGray: false,
  leftIcon: null,
  primary: false,
  white: false,
  small: false,
  svgPrimary: false,
  svgWhite: false,
  valPrimary: false,
  valWhite: false
};

CardItem.displayName = 'CardItem';

export default CardItem;
