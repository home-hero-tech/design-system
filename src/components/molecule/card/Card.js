import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import css from './Card.module.scss';

const Card = ({ className, border, children, ...otherProps }) => {
  const classes = classNames({
    [css['c-card']]: true,
    [css['c-card--borderless']]: !border,
    [className]: !!className
  });

  return (
    <div className={classes} {...otherProps}>
      {children}
    </div>
  );
};

Card.propTypes = {
  className: PropTypes.string,
  border: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

Card.defaultProps = {
  className: '',
  border: true,
  children: null
};

export default Card;
