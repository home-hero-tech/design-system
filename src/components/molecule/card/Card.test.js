import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card from './Card';
import CardItem from '../../atom/card-item/CardItem';

describe('Card', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <Card>
        <CardItem label="Label" value="Value" />
      </Card>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
