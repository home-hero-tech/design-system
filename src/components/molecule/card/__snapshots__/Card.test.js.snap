// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match with the snapshot 1`] = `
<Grid
  fluid={true}
>
  <Row
    className="c-card"
  >
    <ListItem
      col={null}
      ellipsis={false}
      format="DD MMM YYYY [Ã s] HH:mm"
      label="Label"
      primary={false}
      value="Value"
    />
  </Row>
</Grid>
`;
